lucille-core/src/main/java/com/kmwllc/lucille/core/RunDetails.java

```diff
@@ -0,0 +1,177 @@
+package com.kmwllc.lucille.core;
+
+import java.time.Instant;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import com.kmwllc.lucille.core.Runner.RunType;
+import io.swagger.v3.oas.annotations.media.Schema;
+
+@Schema(
+    description = "Details about a specific run, including metadata, start time, and processing stats.")
+public class RunDetails {
+
+  @Schema(description = "Unique identifier for the run.", example = "run123")
+  private final String runId;
+
+  @Schema(description = "Configuration ID for the run.")
+  private String configId;
+
+  @Schema(description = "Start time of the run in ISO-8601 format.",
+      example = "2023-12-20T14:48:00.000Z")
+  private final Instant startTime;
+
+  @Schema(
+      description = "End time of the run in ISO-8601 format. Null if the run has not completed.",
+      example = "2023-12-20T15:48:00.000Z")
+  private Instant endTime;
+
+  @Schema(description = "Number of documents processed during the run.", example = "5000")
+  private long docsProcessed;
+
+  @Schema(description = "Number of errors encountered during the run.", example = "5")
+  private long errorCount;
+
+  @Schema(hidden = true)
+  private transient CompletableFuture<Void> future;
+
+  @Schema(description = "Run result")
+  private RunResult runResult;
+
+  @Schema(description = "Current status")
+  private volatile String status = "new";
+
+  @Schema(description = "Run type")
+  private RunType runType;
+
+  @Schema(description = "Exception of run")
+  private Throwable exception;
+
+
+  public RunDetails(String runId, String configId) {
+    this.runId = runId;
+    this.configId = configId;
+    this.startTime = Instant.now();
+  }
+
+  public String getConfigId() {
+    return configId;
+  }
+
+  public void setConfigId(String configId) {
+    this.configId = configId;
+  }
+
+  public String getRunId() {
+    return runId;
+  }
+
+  public Instant getStartTime() {
+    return startTime;
+  }
+
+  public Instant getEndTime() {
+    return endTime;
+  }
+
+  public void setEndTime(Instant endTime) {
+    this.endTime = endTime;
+  }
+
+  public long getDocsProcessed() {
+    return docsProcessed;
+  }
+
+  public void setDocsProcessed(long docsProcessed) {
+    this.docsProcessed = docsProcessed;
+  }
+
+  public long getErrorCount() {
+    return errorCount;
+  }
+
+  public void setErrorCount(long errorCount) {
+    this.errorCount = errorCount;
+  }
+
+  public CompletableFuture<Void> getFuture() {
+    return future;
+  }
+  
+  public void setFuture(CompletableFuture<Void> future) {
+    this.future = future;
+  }
+
+  public void complete() {
```

Comment:
I posted a comment about removing or renaming this method (along with completeExceptionally) but can't see it anymore. I am confused about having two methods complete() and setEndTime() that do the same thing -- they have different names, one takes an arg and one doesn't.

```
  public void setEndTime(Instant endTime) {
    this.endTime = endTime;
  }
```

Reply:
setEndTime nor complete exist anymore, endTime is set by finalizing with an exception or setting the RunResults